### PR TITLE
make addon controller respect cluster pause

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Make addon controller respect cluster pause.

**Release note**:
```release-note
Addon controller now respects the pause flag on a cluster
```

### CODE REVIEW

**Comment by mrIncompetent** [comment]
/assign @themue 

cc @bashofmann

File: api/pkg/controller/addon/addon_controller.go [hunk 1]
```diff
@@ -270,6 +270,10 @@ func (c *Controller) sync(key string) error {
 		return nil
 	}
 
+	if cluster.Spec.Pause {
```
Comment (themue) [inline api/pkg/controller/addon/addon_controller.go:274]: Would like to see little hint about sense, like for other condition checks. Would also a similar logging make sense?

**Pull request merged**